polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -1149,6 +1164,78 @@ pub mod pallet {
 			MostRecentContext::<T>::insert(&para, context);
 			Ok(())
 		}
+
+		/// Sets the storage for the authorized current code hash of the parachain.
+		///
+		/// This can be useful when we want to trigger `Paras::force_set_current_code(para, code)`
+		/// from a different chain than the one where the `Paras` pallet is deployed.
+		///
+		/// The main reason is to avoid transferring the entire `new_code` wasm blob between chains.
+		/// Instead, we authorize `new_code_hash` with `root`, which can later be applied by
+		/// `Paras::apply_authorized_force_set_current_code(para, new_code)` by anyone.
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as Config>::WeightInfo::authorize_force_set_current_code_hash())]
+		pub fn authorize_force_set_current_code_hash(
+			origin: OriginFor<T>,
+			para: ParaId,
+			new_code_hash: ValidationCodeHash,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			// if one is already authorized, means it has not been applied yet, so we just replace
+			// it.
+			if let Some(already_authorized) = AuthorizedCodeHash::<T>::take(para) {
+				log::warn!(
+					target: LOG_TARGET,
+					"Already authorized code hash: {:?} found for para {:?}, just removing it!",
+					already_authorized, para
+				);
+			}
+
+			// TODO: FAIL-CI - more validations?
+			// do we need to check against `FutureCodeHash`, `CodeHashRef`,
+			// `PastCodeHash`,... code hashes?
```

Comment:
I don't see a reason. `force_set_code_hash` will still do further checks and reject if the authorized code is incorrect.

Reply:
Actually, `do_force_set_current_code_update` or `force_set_current_code` does not do any validations.
So the only validation is here "if code is authorized".

I have the only doubt here:

> > ## Open questions
> 
>     * [ ]  Do we need something like `poke_authorized_code_hash`? E.g. in case that we authorize code hash, but nobody would apply it and the parachain starts working with old/other_new code? Is this possible?